xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/suggest/suggestPages.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+require.config({
+  paths: {
+    'xwiki-selectize': "$xwiki.getSkinFile('uicomponents/suggest/xwiki.selectize.js', true)" +
+      "?v=$escapetool.url($xwiki.version)"
+  }
+});
+
+define('xwiki-suggestPages', ['jquery', 'xwiki-selectize'], function($, utils) {
+  'use strict';
+
+  var getSelectizeOptions = function(select) {
+    return {
+      create: true,
+      load: function(text, callback) {
+        loadPages(text).done(callback).fail(callback);
+      }
+    }
+  };
+
+  var loadPages = function(text) {
+    // ${request.contextPath}/rest/wikis/query/
```

Comment:
The [old page "picker"](https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/FrontendResources/SpecialCSSClasses/#HB2.Behaviorclassnames) that you can obtain using the ``.suggestDocuments`` CSS class is using (see [xwiki.js](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js#L1498)) the following REST URL

    ${request.contextPath}/rest/wikis/__wiki__/search?scope=name&number=10

(which uses database search rather than Solr) You can check for instance the "Go To Page" (Ctrl + G) feature. So I would rather use the same service to get the suggestions, improving it of course if we need to.

The **generic** page picker and the Page **property** picker have different suggestions in may cases:

* when an explicit query is used
* when the implicit query restricts the pages using the class name field
* the generic picker doesn't have "used values"

so I think it's fine that they don't use the same service to get the suggestions and thus that there may be differences in the provided suggestions.

Reply:
We can try to use the existing REST API to fetch suggested documents but I'm afraid that the *generic page picker* won't have the same behavior as the *page property picker* when the latter is used with its default behavior (no explicit query or other restrictions). I'll do some tests.